omod/src/main/java/org/openmrs/module/mksreports/web/controller/MKSReportsManageController.java

```diff
@@ -0,0 +1,144 @@
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.mksreports.web.controller;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.HashMap;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.xml.transform.Result;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.sax.SAXResult;
+import javax.xml.transform.stream.StreamSource;
+
+import org.apache.fop.apps.FOUserAgent;
+import org.apache.fop.apps.Fop;
+import org.apache.fop.apps.FopFactory;
+import org.apache.fop.apps.MimeConstants;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.mksreports.reports.PatientHistoryReportManager;
+import org.openmrs.module.patientsummary.PatientSummaryResult;
+import org.openmrs.module.patientsummary.PatientSummaryTemplate;
+import org.openmrs.module.patientsummary.api.PatientSummaryService;
+import org.openmrs.module.reporting.report.ReportDesign;
+import org.openmrs.module.reporting.report.service.ReportService;
+import org.openmrs.util.OpenmrsClassLoader;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+/**
+ * The main controller.
+ */
+@Controller
+public class MKSReportsManageController {
+	
+	/**
+	 * The path to the style sheet for Patient History reports.
+	 */
+	public static final String PATIENT_HISTORY_XSL_PATH = "patientHistoryFopStylesheet.xsl";
+	
+	/**
+	 * Receives requests to run a patient summary.
+	 * 
+	 * @param patientId the id of patient whose summary you wish to view
+	 * @param summaryId the id of the patientsummary you wish to view
+	 */
+	@RequestMapping(value = "/module/mksreports/patientHistory")
+	public void renderPatientHistory(ModelMap model, HttpServletRequest request, HttpServletResponse response,
+	        @RequestParam("patientId") Integer patientId,
+	        @RequestParam(value = "download", required = false) boolean download,
+	        @RequestParam(value = "print", required = false) boolean print) throws IOException {
+		
+		try {
+			
+			PatientSummaryService patientSummaryService = Context.getService(PatientSummaryService.class);
+			ReportService reportService = Context.getService(ReportService.class);
```

Comment:
Removed the initialization of the services by the context, and injected them by @Autowired annotation rather than using  @SpringBean  annotation, to use this last one I would  need to add uiframework-api dependency